src/chem/modis_aod2ioda.py

```diff
@@ -37,27 +37,38 @@
 
 AttrData = {
     'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
+
+
+DimDict = {
+}
+
+VarDims = {
+    'aerosol_optical_depth': ['nlocs']
 }
 
 
 class AOD(object):
-    def __init__(self, filenames, obs_time, writer):
+    def __init__(self, filenames, obs_time):
         self.filenames = filenames
         self.obs_time = obs_time
-        self.writer = writer
         self.varDict = defaultdict(lambda: defaultdict(dict))
         self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-        self.loc_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-        self.var_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-        self.units = {}
+        self.varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
         self._read()
 
     def _read(self):
         # set up variable names for IODA
         for ncvar, iodavar in obsvars.items():
-            self.varDict[iodavar]['valKey'] = iodavar, self.writer.OvalName()
-            self.varDict[iodavar]['errKey'] = iodavar, self.writer.OerrName()
-            self.varDict[iodavar]['qcKey'] = iodavar, self.writer.OqcName()
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'unitless'
+            self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'unitless'
```

Comment:
After some discussion, we settled on "unitless" being the units name for categorical variables such as qc flags and enumerations. Whereas for physical quantities that happen to be without units, we settled on using '1' as the units name. We are doing this so we can distinguish between variables that can be used in mathematical calculations (units = "1") from those that cannot (units = "unitless").

I'm thinking that aerosol_optical_depth is a unit-less physical quantity that can be used in mathematical formulas. If that is correct, then please set the units to '1'.

Also, please add in the OqcName version of the variables and set their units to "unitless". Eg,
```
self.varAttrs[iodavar, iconv.OqcName()]['units'] = 'unitless'
```

Thanks!

Reply:
Still need to add the units to the "OqcName" version of the variable as shown above. Thanks!

Reply:
Thanks for adding this in!